builtin/providers/azurerm/resource_arm_app_service_plan.go

```diff
@@ -0,0 +1,132 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/web"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppServicePlan() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServicePlanCreateUpdate,
+		Read:   resourceArmAppServicePlanRead,
+		Update: resourceArmAppServicePlanCreateUpdate,
+		Delete: resourceArmAppServicePlanDelete,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tier": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"maximum_number_of_workers": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppServicePlanCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	AppServicePlanClient := client.appServicePlansClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Server Farm creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	tier := d.Get("tier").(string)
+
+	sku := web.SkuDescription{
+		Name: &tier,
```

Comment:
@tombuildsstuff do you mean that I have to pass in the Size and Tier as inputs to the provider? At the moment I am only passing in the service tier, e.g. "S0" which determines the tier.